app/Observers/EntryObserver.php

```diff
@@ -28,11 +29,13 @@ public function created(Entry $entry)
      */
     public function updated(Entry $entry)
     {
-        $character = $entry->character;
-        if ($entry->isDirty('levels')) {
-            $levelDelta = $entry->levels - $entry->getOriginal('levels');
-            $character->level += $levelDelta;
-            $character->save();
+        if ($entry->type != Entry::TYPE_DM || !is_null($entry->character)) {
+            $character = $entry->character;
+            if ($entry->isDirty('levels')) {
+                $levelDelta = $entry->levels - $entry->getOriginal('levels');
```

Comment:
If the DM entry already had a level that was greater than zero but did not have a `character_id` attached to it, then the `levelDelta` would be null and no levels would be added to the newly attached character. I think this calls for a more robust set of rules in the `app/Http/Requests/EntryStoreRequest.php` file or more checks in this function. Let me know what you think.

Reply:
Good catch, I'll fix that right now.

Reply:
I don't think that fixes the issue I was talking about. If a DM Entry got created with a level and no character, sending in a request that updates the level and attaches that DM Entry to a character would only add the difference in levels to that new character.

Reply:
You're right. I rewrote the update method on the observer to catch this edge case.

Reply:
Perfect, I think everything looks good. Nice work! I'm happy we have all cases sorted out.